libraries/abstractions/platform/freertos/iot_network_freertos.c

```diff
@@ -679,18 +685,22 @@ IotNetworkError_t IotNetworkAfr_Destroy( void * pConnection )
     {
         /* Set the flag specifying that the connection is destroyed. */
         ( void ) xEventGroupSetBits( ( EventGroupHandle_t ) &( pNetworkConnection->connectionFlags ),
-                                     _FLAG_CONNECTION_DESTROYED );
+                                     _FLAG_RECEIVE_TASK_CONNECTION_DESTROYED );
     }
     else
     {
-        /* If a receive task was created, wait for it to exit. */
-        if( pNetworkConnection->receiveTask != NULL )
+        /* As this function should be called ONLY called after the connection is closed,
+         * the receive task should have already exited. */
+        if( pNetworkConnection->receiveCallback != NULL )
```

Comment:
What is the reason why we check the `receiveCallback` is not `NULL` here?

Reply:
It is possible that the `receiveCallback` is not set by the application, in which case the network receive task will not be spawned by this network abstraction implementation. Refer [here](https://github.com/aws/amazon-freertos/blob/201912.00/libraries/abstractions/platform/freertos/iot_network_freertos.c#L441-L471) for the function that sets the receive callback.

Reply:
If the application sets `receiveCallback` to `NULL`, won't there be a null-pointer dereference when `_networkReceiveTask` is called? Should the same check also be added there as well?

Reply:
The `_networkReceiveTask` is created only when the `receiveCallback` callback is non-NULL. We can add an assert check in the receive task for the validity of the receive callback